spec/features/workflow_service_creation_spec.rb

```diff
@@ -11,15 +11,17 @@
       roles: []
     )
   end
-  before do
-    allow_any_instance_of(ItemsController).to receive(:current_user)
-      .and_return(current_user)
-    allow_any_instance_of(CatalogController).to receive(:current_user)
-      .and_return(current_user)
+  before :each do
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(current_user)
+    @druid = 'druid:qq613vj0238' # a fixture Dor::Item record
   end
-  scenario 'redirect and display on show page' do
-    visit add_workflow_item_path 'druid:qq613vj0238'
-    click_button 'Add'
+
+  scenario 'redirect and display on show page - with JS', js: true do
```

Comment:
That's part of the puzzle.  Weirdness stopped when it was turned on.  In this version of the spec, I think it is required for `find` to get the CSS # for button.  That's a guess, I don't know for sure.  I've had problems with `click_button` before.  I'm not entirely convinced that this revision is the right way or the best way to do things, I just know that it gave more consistent spec results.